### ISSUE TITLE
FEAT: KL loss 및 training loop 구현

### ISSUE THREAD

**Issue by coouir** [LighTrip/LighTrip-AI#84]
# FEAT: KL loss 및 training loop 구현

## 목적

soft-label classification 학습을 위한 KL divergence loss와 training loop를 구현합니다.

## 작업 내용

- [ ] KL divergence loss 구현
- [ ] optimizer 설정
- [ ] scheduler 설정
- [ ] training loop 구현
- [ ] validation loop 구현
- [ ] checkpoint 저장 로직 구현
- [ ] logging 구현
- [ ] 학습 설정을 config에서 읽도록 수정

## loss 정의

    log_probs = log_softmax(logits, dim=-1)
    target_probs = target_distribution

    loss = KLDivLoss(
        log_probs,
        target_probs,
        reduction="batchmean"
    )

## 기본 학습 설정

    optimizer: AdamW
    learning_rate: 3e-4
    weight_decay: 1e-4
    batch_size: 64 또는 128
    epochs: 15~25

## 기록할 값

- train_loss
- val_loss
- val_ndcg@5
- top1_wcag_pass_rate
- color_distribution

## 산출물

- `src/training/train.py`
- `src/training/losses.py`
- `outputs/checkpoints/`
- `outputs/reports/training_logs/`

## 완료 기준

- [ ] train loop가 1 epoch 이상 정상 실행된다.
- [ ] validation loop가 정상 실행된다.
- [ ] checkpoint가 저장된다.
- [ ] train loss와 val loss가 기록된다.
- [ ] best checkpoint 저장 로직이 정상 동작한다.

---
